### ISSUE TITLE
question about train.py 

### ISSUE THREAD

**Issue by carrierlxk** [Russell91/ReInspect#3]
For the function train in train.py codes, I found that, the author also calculates the loss of test data during training:
if i % solver["test_interval"] == 0:
            net.phase = 'test'
            for _ in range(solver["test_iter"]):
                forward(net, input_gen_test.next(), config["net"], False)
                loss_hist["test"].append(net.loss)
            net.phase = 'train'
so, what is the purpose of calsulating the loss of test data?

**Comment by Russell91** [comment]
This is actually the loss of the validation set, and is a very important process for hyper parameter tuning. If you don't intend to modify the architecture much, you probably won't have to worry about it!

**Comment by carrierlxk** [comment]
got it, thanks for your interpretation.

**Comment by carrierlxk** [comment]
another question is about brainwash_800000.h5, is it the pretained model based on images from brainwash dataset? How can I save  my own pretrained model during running train.py ?

**Comment by Russell91** [comment]
Yes, the pretrained model should be exactly reproducable with the code. By default, the train.py will save to /tmp/reinspect_xxx.h5. You can change this in config.json if you like.
